app/controllers/tasks_controller.rb

```diff
@@ -13,19 +14,23 @@ def index
   # GET /tasks/1.json
   def show
     @task = Task.find(params[:id])
-    render json: @task
+   # render json: @task
   end
 
+
+  def new
+    @task = Task.new
+  end
+
+
   # POST /tasks
   # POST /tasks.json
   def create
-    @task = Task.new(task_params)
 
-    if @task.save
-      render json: @task, status: :created, location: @task
-    else
-      render json: @task.errors, status: :unprocessable_entity
+    if params[:task][:website]
+      parse_webpage(params[:task][:website])
```

Comment:
i believe this should be self.parse_webpage i could be wrong though and you might want move this to the modal